_posts/2020-12-09-its-just-a-monitoring-change.md

```diff
@@ -0,0 +1,34 @@
+---
+layout:     post
+title:      It's Just a Monitoring Change
+date:       2020-12-09 09:00:00
+summary:    Have you ever had a seemingly innocuous change to one system affect another in a catastrophic way? If yes, you might notice a few familiar themes in this write-up. If no, then read it now, before it's too late.
+author:     ols
+category:   Incident Response
+tags:       incident, failure, monitoring, learning
+---
+
+There is somewhat of a trope in technology that suggests changes to monitoring systems are not necessarily held to the same high standard as changes to other, often more complex or revenue-generating, systems and services. The reality is that monitoring systems are complicated beasts, and should not be taken for granted; nor should we be complacent when making changes to what is still a production system. Never again will I utter the words "it's just a monitoring change". Never again will I fill in the Impact field of a change request with these words, because I have learned that even though it might be **_just_** a monitoring change, it can still take down your primary database and render your products unusable. Let's rewind, and add a little context.
+
+Part of our technology stack for backend account services (we're talking super-backend, as in direct connections to the databases) uses an abstraction layer provided by a third-party. It's closed source, but we have a route into them for bug fix and feature requests. One such feature request was to expose metrics to show the count of payments waiting to be fulfilled by one such utility server. As the year is 2020, this was delivered as a Prometheus query exporter application that could be scraped by our Prometheus instances. We could then graph these counts, and raise alerts when the fulfillment queue was growing beyond expectations, giving us additional visibility into problems.
```

Comment:
“As the year is 2020” is this a statement on how the solution for metrics exporters changes annually 😂

Reply:
More a comment on my disdain for Perl scripts returning status codes for Nagios 😉